src/java/org/apache/cassandra/net/StartupClusterConnectivityChecker.java

```diff
@@ -73,56 +80,84 @@ public static StartupClusterConnectivityChecker create(int targetPercent, int ti
      * @return true if the requested percentage of peers are marked ALIVE in gossip and have their connections opened;
      * else false.
      */
-    public boolean execute(Set<InetAddressAndPort> peers)
+    public boolean execute(Set<InetAddressAndPort> peers, Function<InetAddressAndPort, String> getDatacenter)
     {
-        if (targetPercent == 0 || peers == null)
+        if (peers == null)
```

Comment:
Yea that was my idea, that we should always prime the connections but we can just not wait (if you set the local dc option to a really large number then it basically primes and immediately returns).

Reply:
In the latest version I added the ability to disable this by setting a negative timeout. I guess it's probably a good idea to have an opt out option since we're post freeze.